### DISCUSSION TITLE
Allow to add attributes to models

### DISCUSSION THREAD

**Discussion by ccrvlh** [pydantic/pydantic#2933] [Question] [answered]
The use case I currently have is the following:

```python

# This is what comes from a post request, for example.
class UserSchema(BaseModel):
    id: int
    name: str

# But in my database I have:
class UserModel(db.Model):
    id = Column(Integer...)
    name = Column(String...)
    is_active = Column(Boolean)

# "is_active" shouldn't be in the Model (I won't receive it in the API, but I want to add this field, like:

user: UserSchema = .... # the POST request from the API with the type "User"
user.is_active = False

insert = UserModel(**user.dict()) # as I just added `is_active`, the `dict` should also contain it.
db.sesssion.add(insert).

# The `is_active` won't appear anywhere else: documentation, schema, validation when instantiating etc.
```

But Pydantic won't let me add "is_active": `ValueError: "User" object has no field "is_active"`
I understand the core is parsing and validation, but if I'm explicitly adding to the model, it seems that the lack of validation shouldn't be much of an issue. Dataclasses allow it as well, but won't include in the `dict()`, it could just work like `UserSchema.dict(include_additions=True)` or something.  If the strictness of the parsing/validation is crucial, it could be possible to add something like:

```python
class UserSchema(BaseModel):
    id: int
    name: str

    class Config:
         unsafe = True
```

This could denote that the Schema is subject to change (but only if explicitly added. I understand there might be a workaround with private attributes, exclude from model e etc, but it seems to add extra complexity for a simple operation. It could just accept an operation explicitly written in the code. 

EDIT: The other thing that could be done is to add the attribute outside Pydantic (`insert.is_active = False`), or even take the dict out of Pydantic and than add the key to the dict.

So makes sense to have Pydantic to behaviour in strict mode, and make transformations outside. Anyways, it would still be nice to have flexibility.

**Comment by Cielquan** [comment, marked as answer]
Maybe this is what you are looking for:

You can set the `extra` setting to `allow`. The default is `ignore`.

Here is the diff for your example above:
```diff
...
+from pydantic import Extra
...
-class UserSchema(BaseModel):
+class UserSchema(BaseModel, extra=Extra.allow):
    id: int
    name: str
...
```

You can also set the config in the `Config` class inside the model.

See the docs for more info: https://pydantic-docs.helpmanual.io/usage/model_config/#options
